aquadoggo/src/graphql/client/mutation.rs

```diff
@@ -29,66 +30,111 @@ impl Mutation {
         )]
         operation_encoded_param: String,
     ) -> Result<PublishEntryResponse> {
+        let store = ctx.data::<SqlStorage>()?;
+        let tx = ctx.data::<ServiceSender>()?;
+
         // Parse and validate parameters
         let args = PublishEntryRequest {
             entry_encoded: EntrySigned::new(&entry_encoded_param)?,
             operation_encoded: OperationEncoded::new(&operation_encoded_param)?,
         };
 
-        let store = ctx.data::<SqlStorage>()?;
-        store.publish_entry(&args).await.map_err(Error::from)
+        // Validate and store entry in database
+        // @TODO: Check all validation steps here for both entries and operations. Also, there is
+        // probably overlap in what replication needs in terms of validation?
+        let response = store.publish_entry(&args).await.map_err(Error::from)?;
+
+        // Load related document from database
+        // @TODO: We probably have this instance already inside of "publish_entry"?
+        match store
+            .get_document_by_entry(&args.entry_encoded.hash())
+            .await?
+        {
+            Some(document_id) => {
+                let operation = Operation::from(&args.operation_encoded);
+                let operation_id: OperationId = args.entry_encoded.hash().into();
+
+                let verified_operation = VerifiedOperation::new(
```

Comment:
It might be neater to use `VerifiedOperation::new_from_entry(entry_encoded, operation_encoded)` here.

Reply:
Ah, thanks, didn't check if that existed, this is much better!